### PR TITLE
 Support DeleteRange() for transactions

### PR DESCRIPTION
**Pull request by Leviathan1995**
Right now `Transactions` is not supported `DeleteRange()`, this PR still uses `WriteBatch` to support it. Because the pessimistic transaction must lock all of the keys before operating them, so use the `Iterator` to iterate over the range of keys before deleting them.

Signed-off-by: leviathan1995 leviathan0992@gmail.com

### CODE REVIEW

**Comment by sagar0** [comment]
@maysamyabandeh Could you take a look when you get a chance?

File: include/rocksdb/utilities/transaction.h [hunk 1]
```diff
@@ -316,6 +316,14 @@ class Transaction {
                               const SliceParts& key) = 0;
   virtual Status SingleDelete(const SliceParts& key) = 0;
 
+  virtual Status DeleteRange(const ReadOptions& read_options,
```
Comment (ajkr) [inline include/rocksdb/utilities/transaction.h]: We should include the warning comment above the `DeleteRange` declaration in "include/rocksdb/db.h" about the negative performance impact.

Comment (Leviathan1995) [reply include/rocksdb/utilities/transaction.h]: @ajkr I already updated it.

**Comment by maysamyabandeh** [comment]
The lack of delete range in transaction API is an assumption all over the source code. Adding them would require a very thorough inspection of the code base. I do not think there is a strong motivation to pay that cost at this moment.

**Comment by ajkr** [comment]
guess we don't have a plan to move this forward.

**Pull request closed without merging**